analyzers/src/SonarAnalyzer.CSharp/Rules/CertificateValidationCheck.cs

```diff
@@ -71,8 +71,8 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(CheckConstructorParameterSyntax, SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression);
         }
 
-        protected override SyntaxNode FindRootClassOrRecordOrModule(SyntaxNode node) =>
-            base.FindRootClassOrRecordOrModule(node) ?? node.FirstAncestorOrSelf<GlobalStatementSyntax>()?.Parent;
+        protected override SyntaxNode FindRootTypeDeclaration(SyntaxNode node) =>
+            base.FindRootTypeDeclaration(node) ?? node.FirstAncestorOrSelf<GlobalStatementSyntax>()?.Parent;
```

Comment:
The branch with `node.FirstAncestorOrSelf<GlobalStatementSyntax>()` returning `null` is not covered but this is just us playing extra safe in case a new, unsupported type is added in the future. As far as I can tell if the type declaration cannot be found it has to be a `GlobalStatementSyntax` node.